pkg/xml/xml.go

```diff
@@ -0,0 +1,31 @@
+package xml
+
+import (
+	"regexp"
+
+	"github.com/alecthomas/chroma/pkg/doctype"
+	"github.com/dlclark/regexp2"
+)
+
+var (
+	tagRe            = regexp2.MustCompile(`(?ism)<(.+?)(\s.*?)?>.*?</.+?>`, regexp2.None)
+	xmlDeclarationRe = regexp.MustCompile(`(?i)\s*<\?xml[^>]*\?>`)
+)
+
+// MatchString check if a text looks like XML.
+func MatchString(text string) bool {
+	// Check if a doctype exists or if we have some tags.
+	if xmlDeclarationRe.MatchString(text) {
+		return true
+	}
+
+	if matched, _ := doctype.MatchString(text, ""); matched {
+		return true
+	}
+
+	if matched, _ := tagRe.MatchString(text); matched {
```

Comment:
Here the python version only considers the first 1000 characters of the text: https://github.com/pygments/pygments/blob/master/pygments/util.py#L201

Reply:
good catch! just forgot to implement it.